content/blog/2020/07/2020-07-03-github-checks-api-plugin-coding-phase-1.adoc

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: "GitHub Checks API Plugin Project - Coding Phase 1"
+tags:
+- github
+- api
+- plugins
+- gsoc
+- gsoc2020
+author: XiongKezhi
+opengraph:
+  image: /images/gsoc/opengraph.png
+---
+
+== Introduction
+
+This blog post is about our coding phase 1 progress on GSoC project: link:/projects/gsoc/2020/projects/github-checks/[GitHub Checks API Plugin].
+
+The GitHub Checks API is a highly customized way to integrate CI tools to make reports for PRs. 
+It allows users to see CI reports on GitHub pages directly.
+
+image:/images/post-images/2020/07-github-checks-api-plugin-coding-phase-1/github-check-run.png[GitHub Checks API]
+
+What's more exciting is that it can leave annotations on specific lines of code, just as the comments people left while reviewing.
+
+image:/images/post-images/2020/07-github-checks-api-plugin-coding-phase-1/github-check-annotations.png[Annotations]
+
+While on Jenkins' side, the source code view provided by link:https://plugins.jenkins.io/warnings-ng/[Warnings Next Generation Plugin] does pretty much the same thing.
+
+image:/images/post-images/2020/07-github-checks-api-plugin-coding-phase-1/source-view.png[Source Code View]
+
+Utilizing such features through GitHub Checks API, it would make Jenkins more convenient to GitHub users.
+
+== Features from Coding Phase 1
+
+In the past month, our team was mostly working on the general checks API and an implementation for GitHub checks API.
+
+=== General Checks API
+
+Although the general checks API is developed based on the semantic meaning of GitHub Checks API, we still want to prepare it for similar concepts on other platforms like link:https://docs.gitlab.com/ee/api/commits.html#commit-status[Commit Status API] from GitLab.
+Contributions for implementations on these platforms will be welcomed in the future.
+
+=== GitHub Checks API Implmentation
+
+Since we now have implemented the support for GitHub Checks API, plugin developers can easily create a GitHub check run for a GitHub branch source project by consuming our API.
```

Comment:
They cannot, because this functionality has not been released yet. Please either update guidelines or describe how to use it (required for alpha releases as well)